<launch>
      <node name="airsim_pose" pkg="airsim" type="pose.py" respawn="false" output="screen">
            <param name="frame_id" type="string" value="world" />
            <param name="pub_node" type="string" value="airsim/pose" />
            <param name="vehicle_name" type="string" value="airsimvehicle" />
      </node>
      <node name="airsim_pose_tf" pkg="airsim" type="airsim_pose_tf_broadcaster.py" respawn="false" output="screen">
            <param name="frame_id" type="string" value="world" />
            <param name="child_frame_id" type="string" value="base_link" />
            <param name="pose_node" type="string" value="airsim/pose" />
      </node>
      <node name="airsim_imu" pkg="airsim" type="imu.py" respawn="false" output="screen">
            <param name="frame_id" type="string" value="base_link" />
            <param name="pub_node" type="string" value="airsim/imu" />
            <param name="sensor_name" type="string" value="imu" />
            <param name="vehicle_name" type="string" value="airsimvehicle" />
      </node>
      <node name="airsim_lidar" pkg="airsim" type="lidar.py" respawn="false" output="screen">
            <param name="frame_id" type="string" value="base_laser" />
            <param name="pub_node" type="string" value="airsim/lidar" />
            <param name="pub_node_gt" type="string" value="airsim/lidargroundtruth" />
            <param name="sensor_name" type="string" value="lidar" />
            <param name="vehicle_name" type="string" value="airsimvehicle" />
      </node>
      <node pkg="tf" type="static_transform_publisher" name="base_to_laser_broadcaster" args="0 0 0 0 0 0 base_link base_laser 100" />
      <node name="airsim_camera" pkg="airsim" type="camera.py" respawn="false" output="screen">
            <param name="frame_id" type="string" value="base_camera" />
            <param name="camera_2_active" type="int" value="1" />
            <param name="camera_3_active" type="int" value="1" />
            <param name="camera_index_1" type="string" value="0" />
            <param name="camera_index_2" type="string" value="0" />
            <param name="camera_index_3" type="string" value="0" />
            <param name="camera_type_1" type="string" value="Scene" />
            <param name="camera_type_2" type="string" value="Segmentation" />
            <param name="camera_type_3" type="string" value="DepthPerspective" />
            <param name="topic_name_1" type="string" value="airsim/camera/scene" />
            <param name="topic_name_2" type="string" value="airsim/camera/segmentation" />
            <param name="topic_name_3" type="string" value="airsim/camera/depth" />
            <param name="camera_height" type="int" value="540" />
            <param name="camera_width" type="int" value="960" />
            <param name="camera_frequency" type="int" value="10" />
             <param name="max_depth_distance" type="int" value="10" />
      </node>
      <node pkg="tf" type="static_transform_publisher" name="base_to_camera_broadcaster" args="0 0 0 0 0 0 base_link base_camera 100" />
      <node pkg="rosbag" type="record" name="rosbag_record" args="record -o /home/airsim/airsim_all_benchmark -a"/>
</launch>